website/content/docs/waypoint-hcl/variables/input.mdx

```diff
@@ -0,0 +1,353 @@
+
+### Input Variable Documentation
+
+[inpage-description]: #input-variable-documentation
+
+To aid in documentation of what defined variables are expected to be used for,
+you can briefly describe the purpose of each variable using the optional
+`description` argument:
+
+```hcl
+variable "image_tag" {
+  type        = string
+  description = "The tag assigned to built Docker images."
+}
+```
+
+The description should concisely explain the purpose of the variable and what
+kind of value is expected.
+
+## Using Input Variable Values
+
+Within the waypoint.hcl that declared a variable, its value can be accessed from
+within [expressions](/docs/waypoint-hcl/syntax/expressions) as `var.<NAME>`,
+where `<NAME>` matches the label given in the declaration block:
+
+-> **Note:** Input variables are _created_ by a `variable` block, but you
+_reference_ them as attributes on an object named `var`.
+
+```hcl
+  deploy {
+    use "docker" {
+      service_port = var.port
+    }
+  }
+```
+
+The value assigned to a variable can only be accessed within the `app` stanza.
+[Top-level parameters](/docs/waypoint-hcl#top-level-parameters), such as the
+project or app names, cannot reference variable values.
+
+## Assigning Values to Custom Input Variables
+
+When variables are declared in your configuration, they can be set in a
+number of ways. Waypoint loads variables in the following order, with
+later sources taking precedence over earlier ones:
+
+- In the project settings via [the Waypoint server's UI](https://www.waypointproject.io/docs/server#waypoint-server).
+- Automatically loaded variable definition files, named with the pattern
+  `*.auto.wpvars`.
+- As environment variables.
+- In variable definitions (`.wpvars`) files specified with the `-var-file` command
+  line option.
+- Individually, with the `-var` command line option.
+
+The following sections describe these options in more detail.
+
+### Variables on the Command Line
+
+To specify individual variables on the command line, use the `-var` option
+when running the `waypoint up` or specific stage commands (e.g. `waypoint build`).
+You can use the `-var` option multiple times in a single command to set several
+different variables.
+
+```
+waypoint up -var="port=8080"
+waypoint up -var='networks=["internal","dev-test"]' -var="port=8080"
+waypoint up -var='network_map={"internal":8080,"dev-test":9701}'
+```
+
+The above examples show appropriate syntax for Unix-style shells, such as on
+Linux or macOS.
+If you're on Windows, we recommend using the Windows Command Prompt (cmd.exe).
+When you pass a variable value to Waypoint from the Windows Command Prompt,
+use double quotes " around the argument:
+
+```
+waypoint up -var "name=value"
+```
+
+If your intended value includes literal double quotes then you'll need to
+escape those with a backslash:
+
+```
+waypoint up -var "name=va\"lue"
+```
+
+PowerShell on Windows cannot correctly pass literal quotes to external programs,
```

Comment:
Hm, not sure if this paragraph was taken from some where but you can totally quote within quotes on powershell: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.1 You should be able to escape a single quote with a backtick for powershell to not interpret it as "quotes".

Reply:
I did lift this from other docs of ours. I think I'll just remove this sentence for now.